federated_content_connector/filters/pipeline.py

```diff
@@ -11,6 +12,48 @@
 from federated_content_connector.models import CourseDetails
 
 
+COURSE_DETAILS_FIELDS = [
+    'id',
+    'course_key',
+    'course_type',
+    'product_source',
+    'start_date',
+    'end_date',
+    'enroll_by',
+]
+
+CourseDetailsData = namedtuple(
+    'CourseDetailsData',
+    COURSE_DETAILS_FIELDS,
+)
+
+
+def generate_course_details_data(course_key):
+    """
+    Given a course_key, generates an instance of ``CourseDetailsData``,
+    populated with data read from either a ``CourseDetails`` model instance,
+    or a dictionary of data returned by ``get_course_data()``, which is implemented
+    as an HTTP call to the course-discovery service.
+    """
+    model_instance = CourseDetails.objects.filter(id=course_key).first()
+    if model_instance:
+        return CourseDetailsData(**{
+            field_name: getattr(model_instance, field_name, None)
+            for field_name in COURSE_DETAILS_FIELDS
+        })
+
+    course_key_str = '{}+{}'.format(course_key.org, course_key.course)
+    course_data_dict = get_course_data(course_key_str, ['course_type', 'product_source'])
```

Comment:
[curious/sanity check] Does the `start_date` need to be passed to `get_course_data` to cover the `CreateApiRenderEnrollmentStep` pipeline's usage of `course_details.start_date`? I guess a similar question for `end_date` in `CreateApiRenderCourseRunStep`, too.

Reply:
oh good call, gotta test this out better still